### ISSUE TITLE
[REF-2810] Error: img is a self-closing tag and must neither have `children` nor use `dangerouslySetInnerHTML`.

### ISSUE THREAD

**Issue by masenf** [reflex-dev/reflex#3273]
**Describe the bug**
When a user makes a simple mistake, and passes the image `src` as a child, this causes a browser error.

Instead Reflex should catch this at compile time and throw an actionable error in the terminal, or just assume what the user meant and apply the child as the `src`

**To Reproduce**

```python
rx.image("/landing/hero.webp", alt="Reflex Logo", width="100%", height="auto"),
```

OOOPS, meant to write it as

```python
rx.image(src="/landing/hero.webp", alt="Reflex Logo", width="100%", height="auto"),
```

**Screenshots**
Poor user experience 😭

![image](https://uploads.linear.app/9513bc1c-787d-499b-8c86-46952ab7013c/af98ca6c-39a7-4623-b034-bca3bd113e94/57be3345-1274-44a1-8998-cb04c8f6b997?signature=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJwYXRoIjoiLzk1MTNiYzFjLTc4N2QtNDk5Yi04Yzg2LTQ2OTUyYWI3MDEzYy9hZjk4Y2E2Yy0zOWE3LTQ2MjMtYjAzNC1iY2EzYmQxMTNlOTQvNTdiZTMzNDUtMTI3NC00NGExLTg5OTgtY2IwNGM4ZjZiOTk3IiwiaWF0IjoxNzE1MzY0NTQxLCJleHAiOjE3NDY5MDA1NDF9.BaAFEwJYU1TGvmsMfj5srbvKYiX8EiGpNTbudBoKVVE)

**Specifics (please complete the following information):**

* Python Version: 3.11.8
* Reflex Version: 0.5.0a1
* OS: macOS 14
* Browser (Optional): Chrome

<sub>From [SyncLinear.com](https://synclinear.com) | [REF-2810](https://linear.app/reflex-dev/issue/REF-2810/error-img-is-a-self-closing-tag-and-must-neither-have-children-nor-use)</sub>

**Comment by masenf** [comment]
@Yummy-Yums i think you would want to override the `create` classmethod instead of `__init__`.

It's the call to `super.__init__` that's triggering the exception, so if you call that before your logic, then the exception will be raised before you get a chance to do anything.

**Comment by Yummy-Yums** [comment]
@masenf applying the child as src still throws a browser error but works fine in the backend , so i guess throwing the TypeError in the `else` block for user to handle is a better option. is this fine?

![image](https://github.com/reflex-dev/reflex/assets/77977520/0c12b7be-d36e-4650-ac7a-f71832ee1f0d)

**Comment by masenf** [comment]
in the else block, why not

```python
return super().create(src=children[0], **props)
```

**Comment by masenf** [comment]
Fixed via #3307
